torchmetrics/functional/classification/auroc.py

```diff
@@ -89,6 +90,21 @@ def _auroc_compute(
     else:
         if mode != DataType.BINARY and num_classes is None:
             raise ValueError('Detected input to `multiclass` but you did not provide `num_classes` argument')
+
+        # If one or more classes has 0 observations, we should exclude them, as its weight will be 0 anyway
+        elif average == AverageMethod.WEIGHTED and len(torch.unique(target)) != num_classes:
```

Comment:
```suggestion
        if average == AverageMethod.WEIGHTED and len(torch.unique(target)) != num_classes:
```
the above ends with `raise`

Reply:
Good point, will fix!